compiler/ballerina-lang/src/main/java/io/ballerina/compiler/api/impl/ReferenceFinder.java

```diff
@@ -816,6 +816,12 @@ public void visit(BLangInvocation invocationExpr) {
         find(invocationExpr.annAttachments);
         find(invocationExpr.restArgs);
 
+        // If the invocation is undefined, required and rest args won't be set. Need to visit argExprs then.
+        if (invocationExpr.requiredArgs.isEmpty() && invocationExpr.restArgs.isEmpty() &&
```

Comment:
Should we use the ``argExprs` at all? This is because the particular field is coming from an old implementation and it is not recommended to use this field even in the phases such as `typeChecker` and `desgugar`.

Reply:
@nadeeshaan I was thinking the same and wanted to ask @pubudu91 if we can populate requiredArgs or restArgs in this scenario. +1 for the suggestion.